clients/python/girder_client/__init__.py

```diff
@@ -618,15 +618,24 @@ def listFile(self, itemId, limit=None, offset=None):
             'id': itemId,
         }, limit=limit, offset=offset)
 
-    def createItem(self, parentFolderId, name, description='', reuseExisting=False):
+    def createItem(self, parentFolderId, name, description='', reuseExisting=False,
+                   metadata=None):
         """
         Creates and returns an item.
+
+        :param parentFolderId: the folder this item should be created in.
+        :param name: the item name.
+        :param description: a description of the item.
+        :param reuseExisting: whether to return an existing item if one with
+            same name already exists.
+        :param metadata: JSON metadata string to set on item.
```

Comment:
What do you think about making this take a dict rather than serialized json?

Reply:
I don't have a problem with that, just trying to be consistent with existing functions, such as [setFolderAccess](https://github.com/girder/girder/blob/master/clients/python/girder_client/__init__.py#L779)

Reply:
I see.  I don't have much experience using girder_client.  Consistency is important, but it would be good to migrate to a more pythonic API going forward.  Perhaps we can make an issue to jsonify these kinds arguments if the value passed is not a string.  For now, this works for me.

Reply:
I agree it would be much more intuitive if these were dicts.

Reply:
I'd be in favor of testing the type, and if it's an instance of `six.string_types` we can pass it as is, otherwise we can call `json.dumps` on it. We could do the same for `setFolderAccess`.

Reply:
I can do that

Reply:
Done